gosdk/init.go

```diff
@@ -0,0 +1,211 @@
+package gosdk
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/ledgerwatch/erigon-lib/kv"
+	"github.com/ledgerwatch/erigon-lib/kv/mdbx"
+	mdbxlog "github.com/ledgerwatch/log/v3"
+	"github.com/rs/zerolog/log"
+
+	"github.com/0xAtelerix/sdk/gosdk/apptypes"
+	"github.com/0xAtelerix/sdk/gosdk/txpool"
+)
+
+type InitConfig struct {
+	ChainID     uint64 // Defaults to 42
+	DataDir     string // Defaults to /data
+	EmitterPort string // Defaults to :9090
+
+	RequiredChains []uint64 // External chain IDs to wait for
+
+	PrometheusPort string               // Optional, leave empty to disable
+	ValidatorID    string               // Optional, for multi-validator metrics/logs
+	CustomTables   kv.TableCfg          // Optional, merged with default tables
+	CustomPaths    *AppchainCustomPaths // Optional, overrides DataDir-derived paths
+}
+
+// AppchainCustomPaths overrides default directory paths.
+// Optional - if not set, paths are derived from DataDir.
```

Comment:
Custom? Why it's in the name?

Reply:
Because these are custom paths for DBs, If provided then they will be used to create DBs and for R/W from them
If not provide then default paths derived via SDK will be used
I thought custom word is better to differentiate from default paths